src/test/java/io/vertx/test/core/FutureTest.java

```diff
@@ -205,6 +206,13 @@ public void testFailurFutureWithNullFailure() {
   }
 
   @Test
+  public void testOrElse() {
+    assertEquals("bye", Future.future().orElse("bye"));
```

Comment:
Throwing a ISE would make it almost impossible to use. I think there are 2 case:

* The user want to retrieve this default result if not completed
* The user want this default result only on failure

As it's 2 different use cases, I would rather use 2 different methods explicitly named (to avoid the `orElse` confusion).